plugins/modules/ipaconfig.py

```diff
@@ -0,0 +1,441 @@
+  become: true
+  tasks:
+    - name: return current values of the global configuration options
+      ipaconfig:
+        ipaadmin_password: password
+      register: result
+    - name: display default login shell
+      debug:
+        msg: '{{result.config.ipadefaultloginshell[0] }}'
+
+    - name: set ipadefaultloginshell and ipamaxusernamelength
+      ipaconfig:
+        ipaadmin_password: password
+        ipadefaultloginshell: /bin/bash
+        ipamaxusernamelength: 64
+'''
+
+RETURN = '''
+config:
+  description: Dict of all global config options
+  returned: When no options are set
+  type: dict
+  options:
+    ipamaxusernamelength:
+        description: maximum username length
+        returned: always
+    ipahomesrootdir:
+        description: default location of home directories
+        returned: always
+    ipadefaultloginshell:
+        description: default shell for new users
+        returned: always
+    ipadefaultprimarygroup:
+        description: default group for new users
+        returned: always
+    ipadefaultemaildomain:
+        description: default e-mail domain
+        returned: always
+    ipasearchtimelimit:
+        description: maximum amount of time (seconds) for a search
+        returned: always
+    ipasearchrecordslimit:
+        description: maximum number of records to search
+        returned: always
+    ipausersearchfields:
+        description: comma-separated list of fields to search in user search
+        type: list
+        returned: always
+    ipagroupsearchfields:
+        description: comma-separated list of fields to search in group search
+        type: list
+        returned: always
+    ipamigrationenabled:
+        description: Enable migration mode
+        type: bool
+        returned: always
+    ipagroupobjectclasses:
+        description: default group objectclasses (comma-separated list)
+        type: list
+        returned: always
+    ipauserobjectclasses:
+        description: default user objectclasses (comma-separated list)
+        type: list
+        returned: always
+    ipapwdexpadvnotify:
+        description: number of days's notice of impending password expiration
+        returned: always
+    ipaconfigstring:
+        description: extra hashes to generate in password plug-in
+        type: list
+        returned: always
+    ipaselinuxusermaporder:
+        description: order in increasing priority of SELinux users
+        returned: always
+    ipaselinuxusermapdefault:
+        description: default SELinux user when no match is found in map rule
+        returned: always
+    ipakrbauthzdata:
+        description: default types of PAC supported for services
+        type: list
+        returned: always
+    ipauserauthtype:
+        description: default types of supported user authentication
+        returned: always
+    ipadomainresolutionorder:
+        description: list of domains used for short name qualification
+        returned: always
```

Comment:
The same comment on naming convention (where CLI variable names are prefered) apply to these.